### PR TITLE
bump version + prepare release notes for v0.4.0

### PR DESCRIPTION
**Pull request by casparvl**

### CODE REVIEW

File: RELEASE_NOTES [hunk 1]
```diff
@@ -1,6 +1,36 @@
 This file contains a description of the major changes to the EESSI test suite.
 For more detailed information, please see the git log.
 
+v0.4.0 (26 September 2024)
+---------------------
```
Comment (boegel) [inline RELEASE_NOTES:5]:
```suggestion
--------------------------

File: RELEASE_NOTES [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+* PyTorch (with torchvision) (#130)
+* LAMMPS (#131, #169)
+* CP2K (#133)
+
+Bug fixes:
+* Find and report duplicate modules (#167)
+* Specify memory as an extra resource in all ReFrame configs (#180)
+* Make req_memory_per_node hook print a warning if memory resource is not configured in ReFrame config (#182)
+* No longer test import of eessi.testsuite.utils directly in CI, as it fails (by design) for the ReFrame 4.6.2 version deployed in EESSI (#183)
+* Update scorecards action to fix scorecards failure (#184)
```
Comment (boegel) [inline RELEASE_NOTES:20]: This is internal business, I wouldn't mention that in the release notes

**Review by boegel — CHANGES_REQUESTED** [review]

File: RELEASE_NOTES (hunk 2 above)
Comment (casparvl) [reply RELEASE_NOTES]: Removed

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**